freqtrade/strategy/interface.py

```diff
@@ -280,6 +284,24 @@ def custom_stoploss(self, pair: str, trade: Trade, current_time: datetime, curre
         """
         return self.stoploss
 
+    def custom_entry_price(self, pair: str, current_time: datetime, current_rate: float,
+                            **kwargs) -> float:
+        """
+        Custom entry price logic, returning the new entry price.
+
+        For full documentation please go to https://www.freqtrade.io/en/latest/strategy-advanced/
+
+        When not implemented by a strategy, returns None, orderbook is used to set entry price
+        Only called when use_custom_entry_price is set to True.
+
+        :param pair: Pair that's currently analyzed
+        :param current_time: datetime object, containing the current datetime
+        :param current_rate: Rate, calculated based on pricing settings in ask_strategy.
+        :param **kwargs: Ensure to keep this here so updates to this won't break your strategy.
+        :return float: New entry price value if provided
+        """
+        return self.entryprice
```

Comment:
The default should return `current_rate`, not an arbitrarily stored value (which will be wrong if 2 pairs have open orders at the same time).
i'd also probably rename the `current_rate` parameter to to `proposed_rate` - which would align it somewhat to `custom_stake_amount()` - which uses `proposed_stake` as the "pre-calculated stake".